From fbb172d09672ae059d13210ead18a7738313af4f Mon Sep 17 00:00:00 2001
Date: Fri, 9 Mar 2018 13:04:17 -0800
Subject: [PATCH] Bug 793900 - 2.7.5: test failure: 105 - python-bindings.
Bug-Upstream: https://bugzilla.gnome.org/show_bug.cgi?id=793900

--- a/bindings/python/tests/CMakeLists.txt
+++ b/bindings/python/tests/CMakeLists.txt
@@ -1,12 +1,13 @@
 
 IF (WITH_PYTHON)
+  find_path(test_core_dir "unittest_support.py" PATHS ${CMAKE_BINARY_DIR}/common/test-core ${CMAKE_SOURCE_DIR}/common/test-core)
   add_custom_target(test-python-bindings ALL DEPENDS unittest_support gnucash-core-c-build gnucash-core-c-py)
   add_dependencies(check test-python-bindings)
   ADD_TEST(python-bindings ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/runTests.py.in)
   SET_PROPERTY(TEST python-bindings PROPERTY ENVIRONMENT
     GNC_BUILDDIR=${CMAKE_BINARY_DIR}
-    PYTHONPATH=${PYTHON_SYSCONFIG_BUILD}:${LIBDIR_BUILD}/gnucash:${CMAKE_BINARY_DIR}/common/test-core
+    PYTHONPATH=${PYTHON_SYSCONFIG_BUILD}:${LIBDIR_BUILD}/gnucash:${test_core_dir}
   )
 ENDIF()
 
 SET(test_python_bindings_DATA
