Last-Update: 2015-07-10
Forwarded: not-needed
Bug-Upstream: https://bugzilla.gnome.org/show_bug.cgi?id=752203
Author: Dmitry Smirnov <onlyjob@member.fsf.org>
Description: Disable broken test

--- a/src/optional/python-bindings/tests/Makefile.am
+++ b/src/optional/python-bindings/tests/Makefile.am
@@ -38,9 +38,9 @@
 	chmod u+x $@
 
 CLEANFILES = runTests.py
 
-TESTS = runTests.py
+#TESTS = runTests.py
 
 clean-local:
 	rm -f translog.*
 
